src/components/bucket/BucketForm.js

```diff
@@ -63,7 +70,11 @@ function DeleteForm({bid, onSubmit}) {
         <Form
           schema={deleteSchema}
           validate={validate}
-          onSubmit={({formData}) => onSubmit(formData)}>
+          onSubmit={({formData}) => {
+            if (typeof onSubmit === "function") {
```

Comment:
I had to add this check in a few places in the code, because flow doesn't seem to follow code paths very accurately; here for instance, we can't end up with onSubmit being anything else than a function.